docs/root/configuration/secret.rst

```diff
@@ -0,0 +1,146 @@
+
+If a listener server certificates needed to be fetched by SDS remotely, it will NOT be marked as active, its port will not be opened before the certificates are fetched. If Envoy fails to fetch the certificates due to connection failures, or bad response data, the listener will be marked as active, and the port will be open, but the connection to the port will be reset.
+
+Upstream cluster is handled in the similar way, if a cluster client certificates needed to be fetched by SDS remotely, it will NOT be marked as active and it will not be used before the certificates are fetched. If Envoy fails to fetch the certificates due to connection failures, or bad response data, the cluster will be marked as active, it can be used to handle the requests, but the requests routed to that cluster will be rejected.
+
+If a static cluster is using SDS, and it needs to define a SDS cluster (unless google grpc is used which doens't need a cluster), the SDS cluster has to be defined before the static cluster using it.
+
+The connection bewteeen Envoy proxy and SDS server has to be secure. One option is to run the SDS server in the same host and use Unix Domain Socket for the connection. Otherwise it requires mTLS between the proxy and SDS server. In this case, the client certificates for the SDS connection must be statically configured.
+
+SDS server
+----------
+
+A SDS server needs to implement the gRPC service `SecretDiscoveryService <https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/discovery/v2/sds.proto>`_.
+It follows the same protocol as other `xDS <https://github.com/envoyproxy/data-plane-api/blob/master/XDS_PROTOCOL.md>`_
+
+SDS Configuration
+-----------------
+
+:ref:`SdsSecretConfig <envoy_api_msg_auth.SdsSecretConfig>` is used to specify the secret. Its field "name" is a required field. If its "sds_config" field is empty, the "name" field specifies the secret in the bootstrap static_resource :ref:`secrets <envoy_api_field_config.bootstrap.v2.Bootstrap.StaticResources.secrets>`. Otherwise, it specifies the SDS server as :ref:`ConfigSource <envoy_api_msg_core.ConfigSource>`. Only gRPC is supported for the SDS service so its "api_config_source" must specify a "grpc_service".
+
+SdsSecretConfig is used in two fields in :ref:`CommonTlsContext <envoy_api_msg_auth.CommonTlsContext>`. The first field is "tls_certificate_sds_secret_configs" to use SDS to get :ref:`TlsCertificate <envoy_api_msg_auth.TlsCertificate>`. The second field is "validation_context_sds_secret_config" to use SDS to get :ref:`CertificateValidationContext <envoy_api_msg_auth.CertificateValidationContext>`.
+
+Examples one: static_resource
+-----------------------------
+
+This example show how to configurate secrets in the static_resource:
+
+.. code-block:: yaml
+
+  static_resources:
+    secrets:
+      - name: "server_cert"
```

Comment:
Nit: in YAML, you don't normally need `""` quotes unless for some disambiguation.

Reply:
Done